rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -631,6 +663,78 @@ func (r *Layer2Relayer) finalizeBatch(dbBatch *orm.Batch, withProof bool) error
 	return nil
 }
 
+func (r *Layer2Relayer) finalizeBundle(bundle *orm.Bundle, withProof bool) error {
+	dbBatch, err := r.batchOrm.GetBatchByIndex(r.ctx, bundle.EndBatchIndex)
```

Comment:
I think it's better check all the batches of this bundle

Reply:
sounds good. added in [1602759](https://github.com/scroll-tech/scroll/pull/1411/commits/16027592b999feeb22ef85e7c410d40289f0c503).